SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialReceiver.cpp

```diff
@@ -1275,6 +1279,25 @@ TWeakObjectPtr<USpatialActorChannel> USpatialReceiver::PopPendingActorRequest(Wo
 	return Channel;
 }
 
+AActor* USpatialReceiver::FindSingletonActor(UClass* SingletonClass)
+{
+	TArray<AActor*> FoundActors;
+	UGameplayStatics::GetAllActorsOfClass(NetDriver->World, SingletonClass, FoundActors);
+
+	// There should be only one singleton actor per class
+	if (FoundActors.Num() == 1)
+	{
+		return FoundActors[0];
+	}
+	else
+	{
+		UE_LOG(LogSpatialReceiver, Warning, TEXT("Found incorrect number (%d) of singleton actors (%s)"),
```

Comment:
I remember we discussed maybe way back when this should be a crash or something? Or nah? Or an error?

Reply:
Could update to error.